# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/ffvpx/libavcodec/mpegaudiodsp.c
# Commit: 186e2a334f2d
# Full Hash: 186e2a334f2deb37b29f44b28bc8c83babeeb9a4
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-02-09 09:37:17
# Regressor Bug: 1877221
# File Overlap Count: 1
# Description:
#   Bug 1877221 - Pull in new ffmpeg code, reapply patch. r=media-playback-reviewers,chunmin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D199909
# ==============================================================================

diff -r c5920de78d16 -r 186e2a334f2d media/ffvpx/libavcodec/mpegaudiodsp.c
--- a/media/ffvpx/libavcodec/mpegaudiodsp.c	Thu Feb 08 10:39:53 2024 +0000
+++ b/media/ffvpx/libavcodec/mpegaudiodsp.c	Thu Feb 08 10:39:54 2024 +0000
@@ -23,7 +23,6 @@
 #include "libavutil/thread.h"
 #include "mpegaudio.h"
 #include "mpegaudiodsp.h"
-#include "dct.h"
 #include "dct32.h"
 
 static AVOnce mpadsp_table_init = AV_ONCE_INIT;
@@ -81,15 +80,12 @@
 
 av_cold void ff_mpadsp_init(MPADSPContext *s)
 {
-    DCTContext dct;
-
-    ff_dct_init(&dct, 5, DCT_II);
     ff_thread_once(&mpadsp_table_init, &mpadsp_init_tabs);
 
     s->apply_window_float = ff_mpadsp_apply_window_float;
     s->apply_window_fixed = ff_mpadsp_apply_window_fixed;
 
-    s->dct32_float = dct.dct32;
+    s->dct32_float = ff_dct32_float;
     s->dct32_fixed = ff_dct32_fixed;
 
     s->imdct36_blocks_float = ff_imdct36_blocks_float;